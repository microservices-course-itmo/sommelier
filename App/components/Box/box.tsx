import React from 'react'
import * as Styled from './box.styled'
import { Props } from './box.types'

const Box = ({
  pX,
  pY,
  mX,
  mY,
  children,
  borderColor,
  borderRadius,
  backgroundColor,
}: Props) => (
  <Styled.BoxContainer
    pX={pX}
    pY={pY}
    mX={mX}
    mY={mY}
    borderColor={borderColor}
    borderRadius={borderRadius}
    backgroundColor={backgroundColor}
  >
    {children}
  </Styled.BoxContainer>
)

export default Box
