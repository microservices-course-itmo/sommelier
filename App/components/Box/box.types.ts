import { ReactNode } from 'react'
import { theme } from '../../theme'

export const SIZE = {
  xs: 'xs',
  s: 's',
  m: 'm',
  l: 'l',
  xl: 'xl',
} as const

export type BoxProps = {
  pX?: keyof typeof SIZE
  pY?: keyof typeof SIZE
  mX?: keyof typeof SIZE
  mY?: keyof typeof SIZE
  borderRadius?: number
  borderColor?: keyof typeof theme.colors
  backgroundColor?: keyof typeof theme.colors
}

export type Props = BoxProps & { children: ReactNode }
