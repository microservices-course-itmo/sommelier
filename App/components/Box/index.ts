export { Box } from './box'
export { BoxProps } from './box.types'
