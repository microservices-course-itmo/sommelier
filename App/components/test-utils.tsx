import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const themeRender = (component: any) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
