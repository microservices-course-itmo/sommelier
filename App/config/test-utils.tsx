import React, { ReactNode } from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '../theme'

const themeRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
