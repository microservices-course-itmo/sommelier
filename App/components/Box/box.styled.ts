import styled from 'styled-components/native'
import { BoxProps } from './box.types'

export const BoxContainer = styled.View<BoxProps>`
  padding: ${({ theme, pX, pY }) =>
    `${pX ? theme.boxSizes[pX] : 0} ${pY ? theme.boxSizes[pY] : 0}`};
  margin: ${({ theme, mX, mY }) =>
    `${mX ? theme.boxSizes[mX] : 0} ${mY ? theme.boxSizes[mY] : 0}`};
  background-color: ${({ theme, backgroundColor }) =>
    backgroundColor ? theme.colors[backgroundColor] : theme.colors.black};
  ${({ borderRadius }) => !!borderRadius && `border-radius: ${borderRadius}px`};
  border-color: ${({ theme, borderColor }) =>
    borderColor ? theme.colors[borderColor] : theme.colors.black};
`
