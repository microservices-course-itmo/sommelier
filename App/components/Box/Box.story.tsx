import React from 'react'
import { storiesOf } from '@storybook/react-native/dist'
import Box from './box'
import { BufferView, ThemeDecorator } from '../../config/decorators'
import { withKnobs, text, boolean, number, color } from '@storybook/addon-knobs'
import { Text, View } from 'react-native'

const defaultBox = () => (
  <Box
    m={number('m (margin)', 0)}
    mh={number('mh (margin-horizontal)', 0)}
    mv={number('mv (margin-vertical)', 0)}
    ml={number('ml (margin-left', 0)}
    mr={number('mr (margin-right', 0)}
    mt={number('mt (margin-top', 0)}
    mb={number('mb (margin-bottom', 0)}
    p={number('p (padding', 0)}
    ph={number('ph (padding-horizontal', 0)}
    pv={number('pv (padding-vertical', 0)}
    pl={number('pl (padding-left', 0)}
    pr={number('pr (padding-right', 0)}
    pt={number('pt (padding-top', 0)}
    pb={number('pb (padding-bottom', 0)}
    color={color('color', '#ffffff')}
    border={text('border', '1px solid #000000')}
    borderRadius={text('borderRadius', '5px')}
    boxShadow={text('boxShadow', '0px 2px 5px rgba(94, 94, 94, 0.2)')}
  >
    <View style={{ height: number('Height of inner component', 300) }}>
      <Text>{text('Text inside', 'I am box')}</Text>
    </View>
  </Box>
)

storiesOf('Box', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .addDecorator(withKnobs)
  .add('default', defaultBox)
