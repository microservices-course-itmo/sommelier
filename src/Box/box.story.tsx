import React from 'react';
import { text } from '@storybook/addon-knobs';
import Box from './box'

export default {
  title: 'Box',
};

export const boxKnob = () => (
  <Box
  >
    {text("text", 'BOXTEXT asdasdasd')}
  </Box>
);