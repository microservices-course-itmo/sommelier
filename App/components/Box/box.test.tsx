import React from 'react'
import { themeRender } from '../test-utils'
import Box from './box'

describe('Box component', () => {
  it('Component matches snapshot', () => {
    const tree = themeRender(<Box>abc</Box>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  // it('Background color prop assigns correctly', () => {
  //   const { getByText } = themeRender(<Box pt={50}>abc</Box>)
  //   expect(getByText('abc').props.pt).toBe(50)
  // })
})
