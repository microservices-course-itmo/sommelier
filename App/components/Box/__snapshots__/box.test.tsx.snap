// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component Component with all props matches snapshot 1`] = `
<View
  backgroundColor="dark"
  borderColor="white"
  borderRadius={5}
  mX="m"
  mY="l"
  pX="xs"
  pY="s"
  style={
    Array [
      Object {
        "backgroundColor": "#0366d6",
        "borderColor": "#fff",
        "borderRadius": 5,
        "marginBottom": 16,
        "marginLeft": 24,
        "marginRight": 24,
        "marginTop": 16,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      },
    ]
  }
  testID="sommelier-box"
>
  abc
</View>
`;

exports[`Box component Component with only necessary props matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "borderColor": "#000",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
  testID="sommelier-box"
>
  abc
</View>
`;
